### PR TITLE
Shop v1

### PR DESCRIPTION
**Pull request by miketwo**
A very basic implementation of a shop that allows buying a card. Related to #50 . 

Next steps:

- Finish adding colorless cards
- Handle relics
- Handle potions
- Add card removal service
- Add sale prices
- Add discounts and interactions with relics/etc

### CODE REVIEW

File: definitions.py [hunk 1]
```diff
@@ -11,4 +11,36 @@ class EncounterType(StrEnum):
...
+    STARTER ='Starter'
+    UNCOMMON ='Uncommon'
+
+class CardType(StrEnum):
+    ATTACK = 'Attack'
+    SKILL = 'Skill'
+    POWER = 'Power'
+    STATUS = 'Status'
+    CURSE = 'Curse'
+    ENTROPIC = 'Entropic'
```
Comment (Vesper-arch) [inline definitions.py:35]: The types that are defined for the potions were not thought through at all. Those types that you've defined are most likely not the best way of doing it.

Comment (Vesper-arch) [reply definitions.py:35]: Like those were made a while ago and i didn't really know what that was supposed to do anyways.

Comment (miketwo) [reply definitions.py:35]: You're talking about Entropic, Snecko, and Dark Essence? I can remove those. Was just trying to be consistent with what's there.

Comment (Vesper-arch) [reply definitions.py:35]: Ok.

**Review by vesper-arch — APPROVED** [review]

**Pull request merged**